### PR TITLE
Add more log statments

### PR DESCRIPTION
**Pull request by borismattijssen**

### CODE REVIEW

File: pkg/workflow/manager.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ import (
 const (
 	workflowUID                    = "workflow-uid"
 	requeueAfterStatusConflictTime = 500 * time.Millisecond
+	requeueJobstoreNotSyncedTime   = 100 * time.Millisecond
```
Comment (nov1n) [inline pkg/workflow/manager.go]: Was half a second too slow?

File: pkg/workflow/manager.go [hunk 2]
```diff
@@ -207,9 +208,8 @@ func (w *WorkflowManager) syncWorkflow(key string) error {
 
 	// Check if the jobStore is synced yet (initialized)
 	if !w.jobStoreSynced() {
-		time.Sleep(100 * time.Millisecond) // @sdminonne: TODO remove hard coded value
-		glog.Infof("Waiting for job controller to sync, requeuing workflow %v", key)
-		w.queue.Add(key)
+		glog.V(3).Infof("Waiting for job controller to sync, requeuing workflow %v", key)
+		w.enqueueKeyAfter(key, requeueJobstoreNotSyncedTime)
```
Comment (nov1n) [inline pkg/workflow/manager.go]: Gucc

File: pkg/workflow/manager.go [hunk 3]
```diff
@@ -369,11 +370,22 @@ func (w *WorkflowManager) enqueueAfter(obj interface{}, d time.Duration) {
 	return
 }
 
+// enqueueKeyAfter enqueues a workflow key after a given time.
+// enqueueKeyAfter is non-blocking.
+func (w *WorkflowManager) enqueueKeyAfter(key string, d time.Duration) {
```
Comment (nov1n) [inline pkg/workflow/manager.go]: We already have enqueueController function which does something very similar. Any reason we need both?

File: pkg/workflow/manager.go (hunk 1 above)
Comment (borismattijssen) [inline pkg/workflow/manager.go]: This was already in Dario's code

File: pkg/workflow/manager.go (hunk 3 above)
Comment (borismattijssen) [inline pkg/workflow/manager.go]: At some places in the code we don't have a reference to a WF but only it's key.

**Pull request merged**